File: backend/api/schemas/calendar.py
"""Request/response schema for GET /calendar/weeks-in-month.

CLAUDE.md rule 7: the number of weeks in a month (W1-W5) must be derived
from the calendar, not hardcoded to always be 5. This endpoint is what lets
the frontend build the Assigned-Week dropdown's option list without
hardcoding — backend/weekly_planning/calendar_utils.py::weeks_in_month()
already computes this and is already relied on server-side
(backend/configuration/vendor_edits.py's assigned_week validation); this
just exposes the same function over HTTP.
"""
from pydantic import BaseModel


class WeeksInMonthResponse(BaseModel):
    month: str  # "YYYY-MM", echoed back from the request
    weeks: int  # 4 or 5 — how many W1-W5 tags are valid for this month

File: frontend/src/api/calendar.ts
import { api } from './client';

// Backs the Assigned-Week dropdown's option count — CLAUDE.md rule 7: the
// number of weeks in a month must be calendar-derived, never hardcoded to
// always be 5 (e.g. Feb has only 4).
export const getWeeksInMonth = (month: string) =>
  api.get<{ month: string; weeks: number }>(`/calendar/weeks-in-month?month=${month}`);
